components/pages/Homepage/EventCard.tsx

```diff
@@ -0,0 +1,93 @@
+import Image from "next/image";
+import format from "date-fns/format";
+import styles from "../../../styles/Home.module.css";
+import { isSameDay } from "date-fns";
+
+interface Event {
+  title: string;
+  category?: string[];
+  location: string;
+  img?: string;
+  altText?: string;
+  startTime: string;
+  endTime?: string | null;
+  wholeDay?: boolean;
+}
+
+interface EventCardProps {
+  event: Event;
+}
+
+interface TimeProps {
+  startDateTime: string;
+  endDateTime?: string | null;
+  wholeDay?: boolean;
+}
+
+interface CategoryProps {
+  category: string;
+}
+
+const dayFormatString = "d.L.yyyy";
+const dayFormatStringWithoutYear = "d.L.";
+const timeFormatString = "HH:mm";
+
+const Time = ({ startDateTime, endDateTime, wholeDay }: TimeProps) => {
+  const startTimeDate = new Date(startDateTime);
+  const endTimeDate = endDateTime ? new Date(endDateTime) : null;
+  const startTime = format(startTimeDate, timeFormatString);
+  const endTime = endTimeDate ? format(endTimeDate, timeFormatString) : null;
+
+  const shouldShowEndDate = !!endTimeDate && !isSameDay(startTimeDate, endTimeDate);
+
+  return (
+    <div className={styles.times}>
+      <time dateTime={format(startTimeDate, "yyyy-LL-dd")}>
+        {format(startTimeDate, shouldShowEndDate ? dayFormatStringWithoutYear : dayFormatString)}
+      </time>
+      {shouldShowEndDate ? (
```

Comment:
This block is a bit unreadable. Consider if splitting it somehow would increase readability.